File: src/client/java/com/somefrills/chestui/AreaSelectionMenu.java
package com.somefrills.chestui;

import com.somefrills.features.misc.matcher.MatchInfo;
import com.somefrills.misc.Area;
import com.somefrills.misc.Utils;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.text.Style;
import net.minecraft.text.TextColor;

public class AreaSelectionMenu extends ChestUI {
    private final MatchInfo info;

    public AreaSelectionMenu(ChestUI previousScreen, MatchInfo info) {
        super("Select Area", previousScreen);
        this.info = info;
        rebuild();
    }

    @Override
    protected void build() {
        for(Area area : Area.values()) {
            addItem(createChoiceItem(area.getItem(), area.getDisplayName(), area.getColorHex()));
        }
        addItem(createChoiceItem(Items.STRUCTURE_VOID, "None", 0xFF0000));
    }

    private ItemStack createChoiceItem(Item baseItem, String displayName, int colorHex) {
        ItemStack stack = baseItem.getDefaultStack();
        Utils.setCustomName(stack, Style.EMPTY.withColor(TextColor.fromRgb(colorHex)), displayName);
        return stack;
    }

    @Override
    protected void onItemClick(ItemStack stack, int button) {
        String customName = Utils.getPlainCustomName(stack);
        if (customName.equals("None")) {
            info.area = null;
        } else {
            info.area = Area.fromString(customName);
        }
        close();
    }
}

File: src/main/kotlin/com/somefrills/features/misc/glowmob/chestui/AreaSelectionMenu.kt
package com.somefrills.features.misc.glowmob.chestui

import com.somefrills.features.misc.glowmob.MatchInfo
import com.somefrills.misc.Area
import com.somefrills.misc.Area.Companion.fromString
import com.somefrills.utils.plainCustomName
import com.somefrills.utils.setCustomName
import net.minecraft.network.chat.Style
import net.minecraft.network.chat.TextColor
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items

class AreaSelectionMenu(previousScreen: ChestUI?, private val info: MatchInfo) :
    ChestUI("Select Area", previousScreen) {
    init {
        rebuild()
    }

    override fun build() {
        for (area in Area.entries) {
            addItem(createChoiceItem(area.item, area.displayName, area.colorHex))
        }
        addItem(createChoiceItem(Items.STRUCTURE_VOID, "None", 0xFF0000))
    }

    private fun createChoiceItem(baseItem: Item, displayName: String, colorHex: Int): ItemStack {
        val stack = baseItem.defaultInstance
        stack.setCustomName(Style.EMPTY.withColor(TextColor.fromRgb(colorHex)), displayName)
        return stack
    }

    override fun onItemClick(stack: ItemStack?, button: Int) {
        val customName = stack?.plainCustomName ?: return
        if (customName == "None") {
            info.area = null
        } else {
            info.area = fromString(customName)
        }
        onClose()
    }
}
